src/Morphir/TypeScript/AST.elm

```diff
@@ -15,11 +15,17 @@ that we use in the backend.
 
 -}
 
+import Morphir.IR.FQName exposing (FQName)
+import Morphir.IR.Name exposing (Name)
+import Morphir.IR.Path exposing (Path)
+
 
 {-| -}
 type alias CompilationUnit =
     { dirPath : List String
     , fileName : String
+    , packagePath : Path
+    , modulePath : Path
     , typeDefs : List TypeDef
     }
```

Comment:
The Scala AST has an additional `imports` field, with type `List ImportDecl`. I think we should do the same, instead of putting everything in `typeDefs`. It looks wrong to me that the `TypeDef` type below can now include `namespace` and `import` fields, which I don't think are really type definitions at all.

Reply:
I agree that they're not "type definitions", but I do think that all four things belong in the same list. Perhaps it's the name "type definitions" that should change.

#### `ImportStatement` variant
First thing: I should probably be clear that the `ImportStatement` variant isn't for the standard imports at the top of the file. The import statements at the top of the TypeScript file aren't directly represented in the AST at all. They're all generated in the Pretty-Printer. The `ImportStatement` variant is specifically for the strange "`export import`" statements that appear in the top level namespace file. eg these statements from the [sample output](https://github.com/Cynical-Optimist/TypeScriptOutputSamples/blob/dd83ecb6880d53b06a862c5ce566f013af95cdcd/sample-output-morphir-ir/Morphir.ts#L23-L28): 

```
    export import Distribution = Morphir_IR_Distribution
    import Documented = Morphir_IR_Documented
    export import FQName = Morphir_IR_FQName
```

I don't really understand the `export import` syntax being used here, but these statements are taking an imported namespace, changing the name, and (optionally) re-exporting them. They can appear in the main body of a file, and they can appear inside a namespace.

Perhaps we should change the name of this variant to make it clearer? How about `ExportableImport` or `ImportRename`, or `ImportAlias`?

#### Keeping `Namespace` and `ImportStatement` variants
To produce the current output, a module definition needs to contain a list of "named things". So far, those things could be interface declarations, type alias declarations, exportable import declarations, or namespaces.

That's a useful set of things to work with. For instance, it's the exact same set of "named things" that can appear inside a namespace, and the same set of things which may contain a reference to something outside of the module. That means it's the set of things that we need to be able to handle in the function `collectRefsFromTypeDef`.

So means there's a few different places in the code where we want to handle the exact set `( Namespace | TypeAlias | Interface | ImportStatement )`. IMHO that makes it useful to collect them together as variants of one type.

Meanwhile, I'm not sure if there are any places in the code where we want to handle the specific set `( TypeAlias | Interface )`.

Reply:
That seems like a pretty good idea. And you're right: it ought to involve an `AST.ImportStatement` type and that type wouldn't belong with the `typeDef` variants.

However, it would be doing a different job than the existing `ImportStatement` variant (see the other comment below), and I think we still need to keep both. I suggest that the new type would be called `ImportDeclaration` and the original variant can be renamed as `ImportAlias`

Reply:
That sounds good.